platform/windows/display_server_windows.cpp

```diff
@@ -4579,9 +4579,22 @@ DisplayServerWindows::DisplayServerWindows(const String &p_rendering_driver, Win
 	// Init context and rendering device
 #if defined(GLES3_ENABLED)
 
-	if (rendering_driver == "opengl3") {
-		int gl_version = detect_wgl_version();
-		if (gl_version < 30003) {
+	bool fallback = GLOBAL_GET("rendering/gl_compatibility/fallback_to_angle");
+	if (fallback && (rendering_driver == "opengl3")) {
+		Dictionary gl_info = detect_wgl();
+
+		bool force_angle = false;
+
+		Array device_list = GLOBAL_GET("rendering/gl_compatibility/force_angle_on_devices");
+		for (int i = 0; i < device_list.size(); i++) {
+			const Dictionary &device = device_list[i];
+			if (device["vendor"].operator String().to_upper() == gl_info["vendor"].operator String().to_upper() && (device["name"] == "*" || device["name"].operator String().to_upper() == gl_info["name"].operator String().to_upper())) {
```

Comment:
Does it handle gracefully a situation where `device_list` would not contain dictionaries, or dictionaries without the right keys?

Reply:
It seems to be handling it gracefully ("<null>" for the wrong values). Non Dictionary in the Array should be fine, but I'm not sure if reading keys is always safe, will add a check.